source/ptreferential/tests/ptref_ng_test.cpp

```diff
@@ -300,3 +314,71 @@ BOOST_AUTO_TEST_CASE(get_connection) {
     BOOST_REQUIRE_EQUAL(indexes.size(), 1);
     BOOST_REQUIRE_EQUAL(b.data->pt_data->lines.at(*indexes.begin())->uri, "B");
 }
+
+BOOST_AUTO_TEST_CASE(get_disruption_by_tag) {
+    ed::builder b("20180710");
+    b.generate_dummy_basis();
+
+    navitia::apply_disruption(
+        b.disrupt(nt::RTLevel::RealTime, "disrupt_0")
+        .tag("my_tag")
+        .impact()
+        .severity(nt::disruption::Effect::NO_SERVICE)
+        .get_disruption(), *b.data->pt_data, *b.data->meta);
+
+    b.finish();
+    b.data->pt_data->sort_and_index();
+    b.data->pt_data->build_uri();
+    b.data->build_raptor();
+
+    auto indexes = make_query_ng(Type_e::Impact, "disruption.tag=my_tag",
+                                  {}, OdtLevel_e::all, {}, {}, *b.data);
+
+    BOOST_REQUIRE_EQUAL(indexes.size(), 1);
+
+    auto idx = *indexes.begin();
+    BOOST_CHECK_EQUAL( get_disruption_uri(idx, *b.data), "disrupt_0");
+}
+
+BOOST_AUTO_TEST_CASE(get_disruptions_with_multiple_tags) {
+    ed::builder b("20180710");
+    b.generate_dummy_basis();
+
+    navitia::apply_disruption(
+        b.disrupt(nt::RTLevel::RealTime, "disrupt_0")
+        .tag("tag_0")
+        .impact()
+        .severity(nt::disruption::Effect::NO_SERVICE)
+        .get_disruption(), *b.data->pt_data, *b.data->meta);
+    navitia::apply_disruption(
+        b.disrupt(nt::RTLevel::RealTime, "disrupt_1")
+        .tag("tag_1")
+        .impact()
+        .severity(nt::disruption::Effect::NO_SERVICE)
+        .get_disruption(), *b.data->pt_data, *b.data->meta);
+    navitia::apply_disruption(
+        b.disrupt(nt::RTLevel::RealTime, "disrupt_2")
+        .tag("tag_2")
+        .impact()
+        .severity(nt::disruption::Effect::NO_SERVICE)
+        .get_disruption(), *b.data->pt_data, *b.data->meta);
+
+    b.finish();
+    b.data->pt_data->sort_and_index();
+    b.data->pt_data->build_uri();
+    b.data->build_raptor();
+
+    auto indexes = make_query_ng(Type_e::Impact, "disruption.tags(tag_0,tag_2)",
+                                  {}, OdtLevel_e::all, {}, {}, *b.data);
+    BOOST_CHECK_EQUAL(indexes.size(), 2);
+
+    std::vector<std::string> disruption_names;
+    boost::range::transform(indexes, std::back_inserter(disruption_names),
+                            [&b](const navitia::idx_t id) {
+        return get_disruption_uri(id, *b.data);
+    });
+
+    auto expected_disruption_names = {"disrupt_0", "disrupt_2"};
+    BOOST_CHECK_EQUAL_RANGE(disruption_names, expected_disruption_names);
+
```

Comment:
Is it possible to add a query to filter with tag_0 and tag_s another way (disruption.tag=tag_0 or disruption.tag=tag_2) ?

Reply:
:+1: